[WordPress → Jetpack app branding] Update mobile card on home page

SiobhyB opened the pull request:
_Context:_ Following the plans to [refocus the WordPress apps on core features](https://make.wordpress.org/mobile/2022/07/27/refocusing-the-wordpress-app-on-core-features/), we plan to move features that are specific to WordPress.com to the Jetpack app. As such, we will eventually replace all mentions of the WordPress app in Calypso with Jetpack.

#### Proposed Changes

* This PR builds on the work done in https://github.com/Automattic/wp-calypso/pull/68062, which changed the home page's app CTA in mobile browsers so that it links to the Jetpack app. Following this PR, the updated CTA will also display in desktop browsers.
* The conditionals that limited the new CTA to mobile browsers have been removed, and some styles have been slightly tweaked to ensure the CTA matches the design for desktop browsers. 

#### Screenshots

<details>
 <summary>Design / Actual ⤵️</summary>

| Design  | Actual |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/195420746-ca6e9167-dc10-4f7c-a8cf-edb45fd01bbf.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/195420646-984564f9-5b4f-43a5-b29e-634eded3be02.png" width="100%"> |

</details>

<details>
 <summary>Before / After ⤵️</summary>

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/195420503-29d892e0-bb66-4209-874b-e9fe0a47834f.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/195420470-18dfed2d-c4b5-4d8f-ba99-b5d5a5b4dbb9.png" width="100%"> |

</details>

#### Testing Instructions

* Navigate to the customer home page and verify that the new CTA displays in the right sidebar.
* In addition, verify that there have been no changes to the way the CTA displays in mobile browsers.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

SiobhyB committed:
Enable Jetpack branding on all devices

This commit removes the 'isAndroid' and 'isIos' conditionals from the 'displayJetpackAppBranding' variable, so that the branding is enabled on desktop devices too.

SiobhyB committed:
Update link to download app

SiobhyB committed:
Styling tweaks for desktop browsers

SiobhyB committed:
Refactor badge code to avoid unnecessary HTML

Prior to this commit, '<div className="go-mobile__app-badges">' would always render, even if the app badges weren't displayed and that div was empty.

This commit makes some refactors, so that all HTML related to the badges will only render when they're displayed.

SiobhyB commented on client/my-sites/customer-home/cards/features/go-mobile/style.scss:
```diff
@@ -63,6 +63,12 @@
 .go-mobile--jetpack {
+	&.customer-home__card.card {
+		@include break-large {
+			padding-left: 48px;
```
This is 24px (the default padding) + the `padding-left` that's added [here](https://github.com/Automattic/wp-calypso/blob/c29ba2e5d238cc0c2207e56a7636bc987b9cf4b2/client/my-sites/customer-home/cards/actions/quick-links/style.scss#L52-L54) for the Quick Links card (which is located above the mobile card in the sidebar).

The breakpoint has been added as the Go Mobile card is relocated below the Quick Links card on smaller devices, so the extra padding isn't necessary there.

dcalhoun replied:
Good catch. The Figma file doesn't address this directly, as it displays the _Tasks_ without _Quick Links_ in every example. I am uncertain as to the desired alignment, but I did notice that smaller screens technically could use adjustment as well, if desired. 

<img width="376" alt="quick-links-alignment" src="https://user-images.githubusercontent.com/438664/195657988-85127f65-ac49-4a0e-ba87-5765871f0803.png">

SiobhyB replied:
> The Figma file doesn't address this directly, as it displays the Tasks without Quick Links in every example

Ah, good point, I think this added padding was an assumption I'd made based on the card's alignment with the site setup list in Figma. 

@shaunandrews, would you be able to confirm whether we should go with the default padding or the extra padding to align with Quick Links? TIA!

**Default:**

<img width="1082" alt="Screenshot 2022-10-14 at 15 31 18" src="https://user-images.githubusercontent.com/2998162/195872254-cbaf348f-4cf5-414c-94cb-d99252209789.png">

**Added Padding / Aligned with Quick Links:**

<img width="1057" alt="Screenshot 2022-10-14 at 15 32 16" src="https://user-images.githubusercontent.com/2998162/195872414-704def70-322a-44af-ac2f-d39e0b36b597.png">

shaunandrews replied:
It feels better to me when aligned with the **Quick links**. 👍

SiobhyB replied:
Thanks both! I'll go ahead to ship this. :D 🚀

SiobhyB commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -45,23 +47,22 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
 							{ translate( 'Everything you need to build and grow your site from any device.' ) }
 						</h3>
 					</div>
-					<div className="go-mobile__app-badges">
```
Refactored to prevent an empty `<div className="go-mobile__app-badges">` rendering when the app badges don't display.

SiobhyB commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -45,23 +47,22 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
+						</Button>
+					) }
 				</div>
-				{ ! showIosBadge && ! showAndroidBadge && ! isDesktopApp && (
```
I've removed the `! isDesktopApp` conditional for the button, but let me know if there's a reason we still need to hide this on the desktop app.

dcalhoun replied:
I agree that removing `! isDesktopApp` makes sense. With the updated design and other conditionals, there could be a scenario where _no button/link_ is displayed, which would be an odd experience. This change means the desktop link will be displayed if the user is not on a mobile device.

dcalhoun commented on client/my-sites/customer-home/cards/features/go-mobile/index.jsx:
```diff
@@ -28,6 +27,9 @@ export const GoMobile = ( { email, sendMobileLoginEmail } ) => {
 		const showAndroidBadge = ! isIos && ! isDesktop;
+		const showBadge = showIosBadge || showAndroidBadge;
+
+		const badgeStoreName = showIosBadge ? 'ios' : 'android';
```
Nit: since this value is only used once, inlining the conditional on line 53 rather than assigning a constant here might ~improve clarity~ reduce the need to jump around the file while reading.

SiobhyB replied:
Done in https://github.com/Automattic/wp-calypso/pull/69000/commits/089f5b047af19b8ad5e865f5f4c0d7ee66773da6, thanks for bringing that perspective!

dcalhoun commented on client/my-sites/customer-home/cards/features/go-mobile/style.scss:
```diff
@@ -102,13 +109,17 @@
 		margin-top: 0;
 		text-decoration: underline;
+		text-underline-offset: 3px;
+		font-size: rem(15px);
```
Does this need a note? 
```suggestion
		font-size: rem(15px); //typography-exception
```

SiobhyB replied:
Ah, yes, that rule had slipped my mind! I've fixed that in https://github.com/Automattic/wp-calypso/pull/69000/commits/4b2c36625acce914e307a0a175fbcf4f4ae62866 and adding the reference for that rule here for posterity:

https://github.com/Automattic/wp-calypso/blob/1fbc67710c8164f0380761e8d3a5fb0d75a5bec8/packages/typography/styles/variables.scss#L31-L34

dcalhoun reviewed (approved):
Well done! I left a few nitpicks that only need to be addressed if desired. 🚀

SiobhyB committed:
Relocate conditional for overall readability

As per the feedback here, this change relocates a variable to avoid developers having to jump around when working on the code, therefore improving overall readability: https://github.com/Automattic/wp-calypso/pull/69000#discussion_r994886718

SiobhyB committed:
Flag that font-size has no associated variable

Following the rule here, we need to add a "typography-exception" note when using font sizes with no corresponding variable: https://github.com/Automattic/wp-calypso/blob/1fbc67710c8164f0380761e8d3a5fb0d75a5bec8/packages/typography/styles/variables.scss#L31-L34

This is to help with discoverability.

SiobhyB committed:
Merge branch 'trunk' into update/customer-home-mobile-card

SiobhyB committed:
Remove unnecessary braces
